File: app/src/test/java/com/beautifulquran/domain/ReciterSyncTest.kt
package com.beautifulquran.domain

import org.junit.Assert.assertEquals
import org.junit.Test

class ReciterSyncTest {
    @Test
    fun `Yasser highlight runs 200 ms ahead`() {
        assertEquals(200L, ReciterSync.highlightAdvanceMs(9))
    }

    @Test
    fun `other reciters keep their source clock`() {
        assertEquals(0L, ReciterSync.highlightAdvanceMs(1))
        assertEquals(0L, ReciterSync.highlightAdvanceMs(19))
    }
}

File: web/src/domain/__tests__/ReciterSync.test.ts
import { describe, expect, it } from 'vitest'
import { highlightAdvanceMs } from '../ReciterSync'

describe('ReciterSync', () => {
  it('runs Yasser highlight 200 ms ahead', () => {
    expect(highlightAdvanceMs(9)).toBe(200)
  })

  it('leaves other reciters on their source clock', () => {
    expect(highlightAdvanceMs(1)).toBe(0)
    expect(highlightAdvanceMs(19)).toBe(0)
  })
})
